src/modules/contract/contract.service.ts

```
@@ -64,54 +78,80 @@ export class ContractService {
+  ): Promise<Contract> {
     if (!file) {
       throw new BadRequestException('No file uploaded');
     }
 
-    // TODO: Implement file parsing logic based on file type
-    // For now, we'll just create a basic contract record
+    const allowedTypes = [
+      'application/pdf',
+      'application/vnd.openxmlformats-officedocument.wordprocessingml.document',
+    ];
+    if (!allowedTypes.includes(file.mimetype)) {
```

Biome (1.9.4)

[error] This variable is used before its declaration.

The variable is declared here:


(lint/correctness/noInvalidUseBeforeDeclaration)